projects/laji-ui/src/lib/modal/modal.service.ts

```diff
@@ -58,7 +61,8 @@ export class ModalService {
     });
     modalComponent.instance.show();
     modalComponent.changeDetectorRef.detectChanges();
-    modalComponent.instance.onHide.subscribe(this.destroyOnHide.bind(this));
+    this.modalHideSub?.unsubscribe();
```

Comment:
Now it doesn't call the `destroyOnHide` function for all modals and some modals are not removed. I tested opening two modals and hiding them (in here http://localhost:3000/observation/list?informalTaxonGroupId=MVL.26) and it left the first modal in the html (it had `display:none` but it's not optimal that it's are not removed).

Reply:
Well, that escalated somewhat... There was a design flaw that I was aware of but didn't want to spend effort on fixing since it was already in the bootstrap API or how we used it at least. See commit message for details.

Reply:
I think it's good idea to remove the references to the latest modal in the modal service. I checked how the hiding inside the component worked in the ngx-bootstrap and it works so that the component can get a reference to its "own" modal by injecting `modalRef` (the modal service provides it when creating the component) and then the component can call `modalRef.hide()`. It would be a nice-to-have feature but not necessary so I think this is good to go